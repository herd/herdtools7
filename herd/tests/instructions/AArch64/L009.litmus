AArch64 L009

{
  0:X2=v; int v[2]={1,2};
  1:X2=w; int64_t w[2]={1,2}; int64_t 1:X0;
}

 P0              | P1              ;
 LDR W0, [X2]    | LDR X0, [X2]    ;
 STR W0, [X2,#4] | STR X0, [X2,#8] ;
forall (0:X0 = 1 /\ 1:X0=1 /\ v[1]=1 /\ w[1]=1)

