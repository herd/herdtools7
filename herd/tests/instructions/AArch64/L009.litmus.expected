Test L009 Required
States 1
0:X0=1; 1:X0=1;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X0=1 /\ 1:X0=1)
Observation L009 Always 1 0
Hash=a4d8718f6ccd4d63c94c6b4e91cad8d0

